### PR TITLE
Add capability to start vmware VMs with cloud init data

### PR DESCRIPTION
**Pull request by mloviska**
As per [VMware image and cloud-init](https://jira.suse.com/browse/TEAM-5569), we need to test
vmware with cloud init. This is possible only in case test backend injects cloud init data in
to the vmx file directly

### CODE REVIEW

**Comment by mloviska** [comment]
Original PR: https://github.com/os-autoinst/os-autoinst/pull/2111
VR: http://kepler.suse.cz/tests/18390# (We will require test case modification, however it proves that cloud-init process was triggered)

**Comment by mloviska** [comment]
Hello @kalikiana, @Martchus! 

Sorry, my previous branch got messed up and after a chit-chat with Cris I have pushed the updated code.

File: consoles/sshVirtsh.pm [hunk 1]
```diff
@@ -522,7 +522,22 @@ __END"
...
-        $self->run_cmd('echo bios.bootDelay = \"10000\" >> /vmfs/volumes/datastore1/openQA/' . $self->name . '.vmx', domain => 'sshVMwareServer');
+        my $vmx = sprintf('/vmfs/volumes/datastore1/openQA/%s.vmx', $self->name);
+
+        # set default boot delay
+        $self->run_cmd("echo bios.bootDelay = \"10000\" >> $vmx", domain => 'sshVMwareServer');
+
+        # inject cloud init metadata and userdata required for the image if there are any
+        my $ci_meta = $bmwqemu::vars{CLOUD_INIT_META};
+        my $ci_user = $bmwqemu::vars{CLOUD_INIT_USER};
+        my $ci_encoding = $bmwqemu::vars{CLOUD_INIT_ENCODING};
```
Comment (okurz) [inline consoles/sshVirtsh.pm:533]: please either die if the according variables are not set or provide a good default.

**Review by okurz — CHANGES_REQUESTED** [review]

File: consoles/sshVirtsh.pm (hunk 1 above)
Comment (mloviska) [reply consoles/sshVirtsh.pm:533]: Sorry, why would you like to add `die`? VMs should be also tested w/o cloud-init as well

Comment (okurz) [reply consoles/sshVirtsh.pm:533]: But if you don't define those variables then undefined variables will be used in string concatenations causing at least Perl warnings if not further weird problems

Comment (mloviska) [reply consoles/sshVirtsh.pm:533]: If I am not mistaken, `undef` is a falsy value, therefore the code should not get into the below condition and attempt to do string concat.

So it should behave like this, or no?

```
No warning in condition
```

```
#!/usr/bin/perl

use strict;
use warnings;

my $a = undef;

unless ($a) {
    print "No warning in condition\n";
}

if ($a) {
    print "Nothing\n";
}
```

Comment (Martchus) [reply consoles/sshVirtsh.pm:533]: Yes, the if-block should do the trick.

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Comment by mloviska** [comment]
If there are no objections, could we possibly merge it?

File: consoles/sshVirtsh.pm (hunk 1 above)
Comment (okurz) [reply consoles/sshVirtsh.pm:533]: Yes, true. Sorry about that. Haven't looked two lines below :D

**Pull request merged**